src/callbacks_step/time_series_dg2d.jl

```diff
@@ -68,6 +70,83 @@ function get_elements_by_coordinates!(element_ids, coordinates, mesh::TreeMesh,
     return element_ids
 end
 
+function get_elements_by_coordinates!(element_ids, coordinates,
+                                      mesh::UnstructuredMesh2D,
+                                      dg, cache)
+    if length(element_ids) != size(coordinates, 2)
+        throw(DimensionMismatch("storage length for element ids does not match the number of coordinates"))
+    end
+
+    # Reset element ids - 0 indicates "not (yet) found"
+    element_ids .= 0
+    found_elements = 0
+
+    # Iterate over all elements
+    for element in eachelement(dg, cache)
+
+        # Iterate over coordinates
+        for index in 1:length(element_ids)
+            # Skip coordinates for which an element has already been found
+            if element_ids[index] > 0
+                continue
+            end
+
+            # Construct point
+            x = SVector(ntuple(i -> coordinates[i, index], ndims(mesh)))
+
+            # Skip if point is not in the current element
+            if !is_point_in_quad(mesh, x, element)
+                continue
+            end
+
+            # Otherwise point is in the current element
+            element_ids[index] = element
+            found_elements += 1
+
+            if mesh.element_is_curved[element] && mesh.polydeg > 1
+                @warn "A time series point inside a curved element may contain errors"
+            end
+        end
+
+        # Exit loop if all elements have already been found
+        if found_elements == length(element_ids)
+            break
+        end
+    end
+
+    return element_ids
+end
+
+# For the `UnstructuredMesh2D` this uses a simple method assuming a convex
+# quadrilateral. It simply checks that the point lies on the "correct" side
+# of each of the quadrilateral's edges (basically a ray casting strategy).
+# OBS! One possibility for a more robust (and maybe faster) algorithm would
+# be to replace this function with `inpolygon` from PolygonOps.jl
+function is_point_in_quad(mesh::UnstructuredMesh2D, point, element)
```

Comment:
```suggestion
@inline function is_point_in_quad(mesh::UnstructuredMesh2D, point, element)
```

Most of the functions are `inline`d, maybe do this also here?

Reply:
Good point! I will investigate this.

Reply:
Same here, this function no longer exists.